com.unity.multiplayer.mlapi/Runtime/Messaging/MessageQueue/MessageQueueProcessor.cs

```diff
@@ -216,7 +216,19 @@ public void ProcessReceiveQueue(NetworkUpdateStage currentStage, bool isTesting)
                         if (!isTesting)
                         {
                             currentQueueItem.UpdateStage = currentStage;
-                            ProcessMessage(currentQueueItem);
+                            try
+                            {
+                                ProcessMessage(currentQueueItem);
+                            }
+                            catch (Exception ex)
```

Comment:
Ok, but do we know what specific exception could happen here, and if so can we catch that one?

And actually, when I look at the `ProcessMessage` impl, it already has a try/catch around the entire body (and also for top-most "Exception").  How could we ever get into the catch block?

Reply:
To answer the second question first: Yes, I realized that as well, and that's why I'm going to back this out. :)

To answer the first question second: `ProcessMessage` can call `InvokeRpc` which goes into user code which can throw literally anything at any time. This try/catch block is to make sure exceptions thrown from user code don't result in things like leaked PooledNetworkBuffers and don't result in other messages in the queue being dropped on the floor because an exception from one RPC halted the processing of the queue.

Reply:
And doesn't break the current stack and never call this:
```
                    //We call this to dispose of the shared stream writer and stream
                    currentFrame.CloseQueue();
```
Which will never properly close the queue and also drop all of the remaining messages to be processed.